scriptmodules/supplementary/audiosettings.sh

```diff
@@ -35,27 +61,101 @@ function gui_audiosettings() {
             1)
                 amixer cset numid=3 0
                 alsactl store
-                printMsgs "dialog" "Set audio output to auto"
+                printMsgs "dialog" "Set audio output to Auto"
                 ;;
             2)
                 amixer cset numid=3 1
                 alsactl store
-                printMsgs "dialog" "Set audio output to headphones / 3.5mm jack"
+                printMsgs "dialog" "Set audio output to Headphones - 3.5mm jack"
                 ;;
             3)
                 amixer cset numid=3 2
                 alsactl store
-                printMsgs "dialog" "Set audio output to HDMI"
+                printMsgs "dialog" "Set audio output to $HDMI"
                 ;;
             4)
+                amixer cset numid=3 3
+                alsactl store
+                printMsgs "dialog" "Set audio output to HDMI 2"
+                ;;
+            M)
                 alsamixer >/dev/tty </dev/tty
                 alsactl store
                 ;;
             R)
                 /etc/init.d/alsa-utils reset
                 alsactl store
+                rm -f "$home/.asoundrc"
                 printMsgs "dialog" "Audio settings reset to defaults"
                 ;;
         esac
     fi
 }
+
+function _bcm2835_alsa_internal_audiosettings() {
+    local cmd=(dialog --backtitle "$__backtitle" --menu "Set audio output." 22 86 16)
+    local options=()
+    local card_index, card_label
+
+    # Get the list of Pi internal cards
+    while read card_no card_label; do
+        options+=("$card_no" "$card_label")
+    done < <(aplay -ql |aplay -ql | sed -En 's/^card ([0-9]+).*\[bcm2835 ([^]]*)\].*/\1 \2/p')
```

Comment:
Minor whitespace issue

Reply:
hm, not sure which one is it, but I removed the 2nd `aplay` pipe - obviously a mistake.

Reply:
I missed the duplicate. Just saw the missing space. Heh. Thanks.